src/Sylius/Component/Order/Model/OrderItemUnit.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Order\Model;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class OrderItemUnit implements OrderItemUnitInterface
+{
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var int
+     */
+    protected $total = 0;
+
+    /**
+     * @var Collection|AdjustmentInterface[]
+     */
+    protected $adjustments;
+
+    /**
+     * @var int
+     */
+    protected $adjustmentsTotal;
+
+    public function __construct()
+    {
+        $this->adjustments = new ArrayCollection();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTotal()
+    {
+        return $this->total;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setTotal($total)
+    {
+        $this->total = $total;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAdjustments($type = null)
+    {
+        if (null === $type) {
+            return $this->adjustments;
+        }
+
+        return $this->adjustments->filter(function (AdjustmentInterface $adjustment) use ($type) {
+            return $type === $adjustment->getType();
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAdjustment(AdjustmentInterface $adjustment)
```

Comment:
Ok, after some discussion in team, we decided not to use trait in this case. It's because in next iteration, we're going to rewrite total calculation logic on all levels (for now in ``Order``, ``OrderItem`` and ``OrderItemUnit``) and it would be necessary to use ``calculateTotal()`` method inside some of adjustments-related functions. Therefore, this methods cannot be implemented in trait and implementing some "adjustements" methods in trait and some in specific class is even worse than code duplication :/